src/main/kotlin/com/exactpro/th2/common/utils/message/MessageBatcher.kt

```diff
@@ -107,8 +107,8 @@ abstract class Batcher<T>(
             batch.addGroups(groupSupplier.invoke())
 
             when (batch.groupsCount) {
-                1 -> future = executor.schedule(::send, maxFlushTime, MILLISECONDS)
                 maxBatchSize -> send()
+                1 -> future = executor.schedule(::send, maxFlushTime, MILLISECONDS)
```

Comment:
Could you please add comment with reminder that order of operations is important and why it is.